### PR TITLE
Package updates

### PR DESCRIPTION
**Pull request by alhardy**
- Update to aspnet core 2.2
- Update swashbuckle to 5 beta
- setup azure pipeline for build and package publishing

### CODE REVIEW

**Comment by chinwobble** [comment]
can we add this to the package too 
```
    public static class OptionsExtensions
    {
        public static IServiceCollection AddConfiguration<TOptions>(this IServiceCollection services, IConfiguration configuration)
            where TOptions : class, new()
        {
            services.Configure<TOptions>(configuration.GetSection(typeof(TOptions).Name));
            return services.AddSingleton(p => p.GetRequiredService<IOptions<TOptions>>().Value);
        }
    }
```

File: sandbox/Sandbox.Api/Sandbox.Api.csproj [hunk 1]
```diff
@@ -17,34 +17,36 @@
...
     <PackageReference Include="Serilog.AspNetCore" Version="2.1.1" />
     <PackageReference Include="Serilog.Filters.Expressions" Version="2.0.0" />
     <PackageReference Include="Serilog.Settings.Configuration" Version="2.6.1" />
-    <PackageReference Include="Serilog.Sinks.ApplicationInsights" Version="2.6.0" />
+    <PackageReference Include="Serilog.Sinks.ApplicationInsights" Version="2.6.2" />
     <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
     <PackageReference Include="Serilog.Sinks.Seq" Version="4.0.0" />
-    <PackageReference Include="Swashbuckle.AspNetCore" Version="$(SwashbuckleVersion)" />
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="5.0.0-beta" />
+    <PackageReference Include="Swashbuckle.AspNetCore.Annotations" Version="5.0.0-beta" />
```
[the remark this answers is not in the document]
Comment (alhardy) [reply sandbox/Sandbox.Api/Sandbox.Api.csproj]: Done

File: src/Prospa.Extensions.AspNetCore.Mvc.Core/StartupFilters/RequireEndpointKeyStartupFilter.cs [hunk 2]
```diff
@@ -32,7 +39,7 @@ void RequireSecretToMetricsAndHealth(IApplicationBuilder app)
                 {
                     var key = ExtractToken(context);
 
-                    if (_endpoints.Any(e => string.Compare(context.Request.Path.Value, e, StringComparison.InvariantCultureIgnoreCase) == 0))
+                    if (DefaultEndpoints.Any(e => context.Request.Path.Value == e))
```
[the remark this answers is not in the document]
Comment (alhardy) [reply src/Prospa.Extensions.AspNetCore.Mvc.Core/StartupFilters/RequireEndpointKeyStartupFilter.cs]: Done

**Comment by alhardy** [comment]
> can we add this to the package too
> 
> ```
>     public static class OptionsExtensions
>     {
>         public static IServiceCollection AddConfiguration<TOptions>(this IServiceCollection services, IConfiguration configuration)
>             where TOptions : class, new()
>         {
>             services.Configure<TOptions>(configuration.GetSection(typeof(TOptions).Name));
>             return services.AddSingleton(p => p.GetRequiredService<IOptions<TOptions>>().Value);
>         }
>     }
> ```

Yes we typically do this, it's a one liner though and at the moment we don't really have a package where this would belong. I'm happy to leave this out for now, @chinwobble ?

**Comment by chinwobble** [comment]
maybe adding it here or adding it to the dotnet template would be good.

**Comment by alhardy** [comment]
> maybe adding it here or adding it to the dotnet template would be good.

I'll add it as an extensions method in the dotnet template

**Pull request merged**